### PR TITLE
output/alert: Add fileinfo to alert

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of draft PR #5017

Redmine issue: [2015](https://redmine.openinfosecfoundation.org/issues/2015)

Describe changes:
- Refactor existing "fileinfo" into a helper function
- Add fileinfo (if `metadata` is configured), to the alert

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/output-json-alert.c [hunk 1]
```diff
@@ -563,6 +563,37 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
...
+            if (ffc != NULL) {
+                File *file = ffc->head;
+                bool isopen = false;
+                while (file) {
+                    if (pa->tx_id == file->txid) {
+                        if (!isopen) {
+                            isopen = true;
+                            jb_open_array(jb, "fileinfo");
+                        }
+                        JsonBuilder *jfb = jb_new_object();
```
Comment (victorjulien) [inline src/output-json-alert.c:579]: why a new object here? We can just pass jb into `JsonFileInfoMulti`?

File: src/output-json.c [hunk 2]
```diff
@@ -152,6 +152,106 @@ json_t *JsonAddStringN(const char *string, size_t size)
     return SCJsonString(tmpbuf);
 }
 
+static void JsonFileInfoInternal(JsonBuilder *js, const File *ff, const bool stored, const bool multi)
+{
+    if (!multi) {
```
Comment (victorjulien) [inline src/output-json.c:157]: I prefer updating the callers to call `jb_open_object` if they need to.

File: src/output-json.c [hunk 3]
```diff
@@ -152,6 +152,106 @@ json_t *JsonAddStringN(const char *string, size_t size)
     return SCJsonString(tmpbuf);
 }
 
+static void JsonFileInfoInternal(JsonBuilder *js, const File *ff, const bool stored, const bool multi)
```
Comment (victorjulien) [inline src/output-json.c:155]: The new funcs are getting a `Eve` prefix instead of `Json`.

File: src/output-json-alert.c (hunk 1 above)
Comment (jlucovsky) [reply src/output-json-alert.c:579]: We're creating elements for an array. I don't think the interface supports what you're asking. Am I misunderstanding @jasonish ?

**Comment by jlucovsky** [comment]
Continued in #5051

**Pull request closed without merging**